enzy_htp/_interface/amber_interface.py

```diff
@@ -12,29 +12,67 @@
 import re
 import shutil
 from pathlib import Path
-from subprocess import CalledProcessError
+from subprocess import CalledProcessError, CompletedProcess, SubprocessError
 from typing import List, Tuple, Union, Dict, Any
+from dataclasses import dataclass
 
 from .base_interface import BaseInterface
-from .handle_types import MolDynParameterizer, MolDynParameter, MolDynStep
+from .handle_types import (
+    MolDynParameterizer, MolDynParameter,
+    MolDynStep,
+    MolDynResult, MolDynResultEgg,)
 from .ncaa_library import search_ncaa_parm_file
 from .gaussian_interface import gaussian_interface
 
 from enzy_htp.core import _LOGGER
 from enzy_htp.core import file_system as fs
-from enzy_htp.core import env_manager as em
+from enzy_htp.core import math_helper as mh
 from enzy_htp.core.job_manager import ClusterJob
-from enzy_htp.core.mol_dyn_result import MolDynResult
-from enzy_htp.core.exception import UnsupportedMethod, tLEaPError
+from enzy_htp.core.exception import UnsupportedMethod, tLEaPError, AmberMDError
 from enzy_htp._config.amber_config import AmberConfig, default_amber_config
 from enzy_htp.structure.structure_io import pdb_io, prmtop_io
+from enzy_htp.structure.structure_constraint import StructureConstraint
 from enzy_htp.structure import (
     Structure,
     Ligand,
     MetalUnit,
     ModifiedResidue,
-    NonCanonicalBase,)
+    NonCanonicalBase,
+    StructureEnsemble)
 from enzy_htp import config as eh_config
+class AmberNCParser():
+    """parser Amber .nc file to StructureEnsemble()
+    Attribute:
+        prmtop_file
+        parent_interface
+    
+    TODO figure out where should these types of structure parser be placed.
+    These parser relies on the interface. They can not be directly used by structure
+    modules such as structure_operations.
+    Can structure_io just be above _interface?? Seems ok."""
+    def __init__(self, prmtop_file: str, interface: BaseInterface):
+        self.prmtop_file = prmtop_file
+        self.parent_interface = interface
+    
+    def get_structure_ensemble(nc_file: str) -> StructureEnsemble:
```

Comment:
I'm assuming this isn't used presently? thses functions shouldnt work as they are now.

Reply:
Yes. They are just an empty shell for development when I get into this.